datasync/kvdbsync/plugin_impl_dbsync.go

```diff
@@ -42,20 +45,27 @@ type infraDeps interface {
 }
 
 // OfDifferentAgent allows accessing DB of a different agent (with a particular microservice label).
-// This method is a shortcut to simplify creating new instance of a plugin
-// that is supposed to watch different agent DB.
-// Method intentionally copies instance of a plugin (assuming it has set all dependencies)
-// and sets microservice label.
-func (plugin /*intentionally without pointer receiver*/ Plugin) OfDifferentAgent(
+// This method creates a new instance of a plugin that is supposed to watch different agent DB.
+// Method copies the existing plugins Deps and sets microservice label.
+func (p *Plugin) OfDifferentAgent(
 	microserviceLabel string, infraDeps infraDeps) *Plugin {
-
+	// Create a new plugin and copy over all the fields from the old one
+	plugin := &Plugin{}
```

Comment:
You should generally try to initialize all fields in single statement if nothing prevents that. 

Imagine you are debugging with debugger and are on breakpoint somewhere, splitting initialization of the fields separately will actually have plugin variable in different state on each step, but when initializing all fields in single statetement it will get fully initialized on the first step.

Also less code 😝